File: src/Tests/Behavioral/InterfaceIntraFunction/InterfaceIntraFunction.go
package main

import "fmt"

type Message struct {
	Text string
}

// Method must be defined at package level, not inside a function!
func (m Message) Print() {
	fmt.Println(m.Text)
}

func main() {
	// Define an interface inside the function
	type Printer interface {
		Print()
	}

	// Instantiate and use the interface
	var p Printer = Message{"Hello, from a function-scoped interface!"}
	p.Print()
}

File: src/Tests/Behavioral/InterfaceIntraFunction/InterfaceIntraFunction.cs
namespace go;

using fmt = fmt_package;

partial class main_package {

[GoType] partial struct Message {
    public @string Text;
}

public static void Print(this Message m) {
    fmt.Println(m.Text);
}

[GoType] partial interface main_Printer {
    void Print();
}

private static void Main() {
    main_Printer p = new Message("Hello, from a function-scoped interface!");
    p.Print();
}

} // end main_package
